### PR TITLE
Expose Nearest.regridder

### PR DESCRIPTION
**Pull request by bjlittle**
This PR exposes the `Nearest.regridder` method, thus hooking up the underlying implementation to the new API.

### CODE REVIEW

**Comment by bjlittle** [comment]
@rhattersley can I tempt you to a bonus 1,000,000 pedant points? :stuck_out_tongue_winking_eye:

File: lib/iris/tests/unit/analysis/test_Nearest.py [hunk 1]
```diff
@@ -103,5 +103,39 @@ def test_nanmask(self):
         self.check_mode('nanmask')
 
 
+class Test_regridder(tests.IrisTest):
+    def check_mode(self, mode=None):
+        scheme = create_scheme(mode)
+
+        with mock.patch('iris.analysis.RectilinearRegridder',
```
Comment (rhattersley) [inline lib/iris/tests/unit/analysis/test_Nearest.py]: I'm not sure it's worth a cool million pedant points (more like half a dozen), but it'd be nice to have a comment here to explain what's going on - i.e. a tweaked version of the [comment for the interpolator](https://github.com/bjlittle/iris/blob/a9631b8c5746777f91a1825dff7eb158c083046d/lib/iris/tests/unit/analysis/test_Nearest.py#L72).

**Comment by rhattersley** [comment]
Bish, bash, bosh... and nearest-neighbour regridding is ready to go. Top work! :clap: 

Add the quick code comment mentioned above and I'd be absolutely delighted to hit the big green button! :grinning:

**Comment by bjlittle** [comment]
@rhattersley Go on, hit it!

**Comment by bjlittle** [comment]
@rhattersley Awesome, thanks!

**Pull request merged**